signer.go

```diff
@@ -0,0 +1,64 @@
+package sdk
+
+import (
+	"context"
+	"encoding/hex"
+	"fmt"
+
+	servicetypes "github.com/pokt-network/poktroll/x/service/types"
+	"github.com/pokt-network/ring-go"
+)
+
+// Signer is used to sign Relay Requests.
+type Signer struct {
+	PrivateKeyHex string
+}
+
+// Note: Sign returns a pointer instead of directly setting the signature on the input relay request.
+// This is done to avoid having an implicit output.
+// Ideally, the function should accept a struct rather than a pointer, and also return an updated struct instead of a pointer.
+//
+// Sign signs the given relay request using the signer's private key
+// and the application's ring signature.
+func (s *Signer) Sign(
+	ctx context.Context,
+	relayRequest *servicetypes.RelayRequest,
+	// TODO_IMPROVE: this input argument should be changed to an interface.
+	app ApplicationRing,
+	queryHeight uint64,
```

Comment:
`queryHeight` is only used by `app.GetRing` which ultimately needs the `SessionEndBlockHeight`.

`SessionEndBlockHeight` is already available in `relayRequest.Meta.SessionHeader.SessionEndBlockHeight`

So it's safe to remove the `queryHeight` argument.

Reply:
Great, thank you! Fixed.